### DISCUSSION TITLE
[Android] Having both "Delete" and "Move to Trash" is confusing and unnecessary

### DISCUSSION THREAD

**Discussion by urbenlegend** [immich-app/immich#14441] [Feature Request]
### The bug

I think having both "Delete" and "Move to Trash" in the Android mobile app is confusing and unnecessary. "Delete" seems to delete the local copy and move the cloud copy to Trash. Meanwhile, "Move to Trash" only deletes the cloud copy and when the photo is permanently deleted from Trash, the app will reupload the local copy. I think this entire behavior is wrong from both a UX and a functionality perspective.

I propose this:
1. Remove the "Delete" button. It's combining two behaviors, "move to trash" and "delete from device" into one confusing button that is unclear, especially when there's already a separate "delete from device" button.
2. Keep the "Move to Trash" button that just moves the cloud copy to trash.
3. When a photo has been moved to Trash and then permanently deleted, only then delete the associated local copy.
4. Keep the "Delete from device" button as is.

This change will prevent the following confusing behaviors:
1. A user "deletes" a photo and then later restores it from Trash. Now he's confused where his local copy is.
2. A user selects "move to trash" on a photo, empties the trash, and then sees the image get re-uploaded to the server from the local copy.

### The OS that Immich Server is running on

Arch Podman-Docker

### Version of Immich Server

v1.121.0

### Version of Immich Mobile App

1.121.0 build.168

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Irrelevant
```

### Your .env content

```Shell
Irrelevant
```

### Reproduction steps

**Delete case**
1. Select a photo and press delete
2. Go into trash and restore that photo
3. Notice that the local copy is gone from the device

**Move To Trash case**
1. Select a photo and press Move to Trash
2. Go into trash and empty it
3. Notice that the local copy is still there in the timeline and later gets backed up.

**Comment by Davide9908** [comment]
I think keeping only the Move to trash and Delete from device buttons will highlight another issue:

If the user wants to delete a photo from both the device and the server he would need to perform the 2 operations separately.
However, if he moves the photo to the trash, the photo will disappear from the app and he will not be able to delete the local file from within the immich app. 

He would need either to use another application to delete the file or remember to use the "Delete from device" button first and then "Move to Trash"

**Comment by relic664** [comment]
I was onboarding new users into immich and found this behavior to be very confusing. The functionality of immich for delete is all there, but the UX is pretty bad. I'm not sure if it was the case when this issue was opened, but there's also a dialog "Allow immich to delete this photo" with allow or deny, and neither seem to do anything perceptible. 

For most "average" users, there's no functional difference between delete and trash; when they delete something normally, stuff just goes to trash. There isn't usually a permanently delete option until you're deleting from trash. 

I think there needs to be clear distinctions between local and cloud storages in dialogs, options, and icons. (Trash in immich being cloud trash can also be confusing).

**Comment by alirz1** [comment]
how about having only two delete options:

DELETE: which moves both local and server side assets to trash. 
Delete from device: only moves local asset to device trash.

**Comment by PDerid** [comment]
I also find the delete process very unclear tbh. I even see 3 delete actions on mobile ("delete from Immich", "delete", "delete from device"). When I see "delete from immich" I expect it only moves the cloud image to the trash, but when I clear the trash, the asset on my local device is also deleted. So basically it has the same behavior as the normal "delete" button. Sometimes I'd like to delete photos on Immich, but keep them on my device.

**Comment by Fmstrat** [comment]
My 2 cents is this goes beyond Android/Mobile, and "delete" should be unified across the server and all mobile devices.

**One delete button to rule them all.**

There are a ton of discussions on deleting from device, from server, on both. In Albums and Search and Listings the delete options can differ. Local files get renamed `.trashed...` and seemingly stick around many devices. Files deleted on the server get re-uploaded by the client.

This reques should expand to unify all delete options across mobile and web into something more UX friendly.

**Suggested process**
- One delete button
-  The same button for all individual and multi-select views
- In settings (mobile and web), add a dropdown select or radios of:
  - Delete moves to Trash on server and all devices
  - Delete moves to Trash on server and deletes from all devices
  - Delete permanently removes from everywhere (big red warning)
  - Ask me every time (default)
- On the "Ask me" popup, it allows the user to save the setting permanently
- If an item in Trash is deleted, it should delete from the server and all devices (no stagnant `.trash` files)

A key element is delete from all logged in devices to eliminate accidental re-uploads and have a unified image list. 

It would also be great if multi-select was modified to make batch deleting easier:

- Hold to select image
- Tap to select another image
- Hold to select that image, and every image between it and the last selected image

**Reply by Mannshoch** [reply]
I like this behavior.  My use case would be, that on android, for around 80% of the cases I move the Pictures to Trash and sometimes Later. I would decide on Immich web to Permanently Remove from all device.
the other cases would be wrong created photos,

**Reply by TitaniumCoder477** [reply]
I was in the process of creating a feature request when I found this conversation and your thread. I was going to ask for a combined button because I only have "Trash" and "Delete from device" on the Android app. I like your solution a lot. It gives flexibility that covers all the difference use cases. Unfortunately, at the moment it looks like either someone removed the third button ("Delete") from the Android app, or it never was in there, because I have to do the multi-select twice and use both buttons to completely eliminate a picture. After my 5 year old nephew's birthday party, this is a tedious chore!

**Comment by isantiago95** [comment]
Hi everyone,
I recently moved from Synology Photos to Immich. At first, I found the three-button approach a bit confusing. After testing with a few images, I eventually understood the behavior and purpose of each button. However, since I had been using Synology Photos for a couple of years, I was already accustomed to their workflow

In Synology Photos, there are only two delete options, and you can choose which one should be the default (see attached image). For me, this approach feels more intuitive than having three separate buttons

![image](https://github.com/user-attachments/assets/0e1d80dc-8d86-4089-8ff0-93fd40700a87)

That said, there is one important difference: Synology Photos does not have a “trash” feature—files are deleted immediately. Based on this, my proposal is to simplify the UI to a single Delete button, while allowing its behavior to be configured in the settings

The configuration options would be similar to Synology Photos, with an additional third option:
	•	Delete from phone and move to trash in the cloud (files would be permanently deleted from the cloud after 30 days)

This way, users can choose the behavior that best fits their use case, keep the interface simple, and still have access to all three delete behaviors—configurable and changeable at any time

**Comment by MrSiO** [comment]
I'm evaluating Immich self hosted (_currently using nextcloud memories_) and the only thing that is slowing me down from migrating is the android app behaviour:

Image deletion is confusing and i lost images while testing.  
It would definitely need a clear "Delete from Server", "Delete from mobile" and "Delete from Everywhere" options.

**Comment by urbenlegend** [comment]
I think three options is needlessly confusing. Delete local copy and Delete Everywhere is enough. Delete from server is weird because now you have specific images that only exist on your device and the app would have to remember to never sync those again.

**Comment by zacharyficker** [comment]
I also think the ‘delete’ button should be the default behaviour in all instances.
Currently it is the prominent function on the app (I'm using iOS) when viewing one asset is ‘delete’.
However, when selecting multiple images, the ‘delete’ button is far down the list, and ‘move to trash’ and ‘delete from device’ are listed earlier.
This is confusing for my less-techy family members, I think the ‘delete’ button that performs both actions should always be the prominent action (if not the only action.

**Comment by urbenlegend** [comment]
~~I personally find the current Delete behavior extremely weird. It deletes local copy while moving the server copy to Trash. It feels like two separate actions combined into one. One is a permanent action (phone) and the other is a recoverable action (server trash). It doesn't feel synchronized to me. If its in trash, I expect it to be a recoverable action both locally and on the server.~~

~~Let's say you're in an area with a crappy connection. You delete a video, it gets wiped from your phone and it gets moved to server trash. Now you decided, shoot, actually that was a mistake, let me restore from Trash. Now it's gotta redownload the file from the server even though from your perspective you thought it was in your local trash.~~

~~Not saying the current delete doesn't have its uses, but I don't think most people expect that as the default behavior. And also, it certainly should not be named Delete, which on most other platforms usually means a permanent unrecoverable removal.~~

**Reply by zacharyficker** [reply]
This is a good point—my perspective comes from iOS where the local copy being ‘deleted’ is always sent to the phone's bin. Perhaps an intermediary bin implementation is worthwhile on Android? Not sure how many distros do/don't support a native bin when photos are deleted by a third-party app.

**Reply by urbenlegend** [reply]
Okay, I was wrong. I wasn't checking the Android trash bin. Turns out the behavior is EVEN more confusing.

Here's the behavior fully mapped out:

**With Sync remote deletions [EXPERIMENTAL] off:**

1. **Move to Trash:** Keeps local file exactly where it is, moves server copy to trash. Restoring from trash simply restores server copy to timeline, your local copy is already there on your phone.
2. **Delete:** Moves local file to Android trash, moves server copy to Immich trash. Restoring from Immich trash restores server copy, but does NOT restore local copy out of Android trash. This is bad because Immich gives you the impression that the local copy was deleted (cloud icon with no checkmark), when in fact, it was not. This is misleading in terms of both privacy AND data storage.

**With Sync remote deletions [EXPERIMENTAL] on:**
1. **Move to Trash:** Moves server copy to Immich trash. Does not immediately move local copy to Android trash until you trigger a server sync by switching to a different app and then reopening Immich. Restoring from trash restores server copy, and you gotta trigger a server sync for it to move local copy out of Android trash.
2. **Delete:** Moves local file to Android trash immediately, moves server copy to Immich trash. Restoring from Immich trash restores server copy, and the local copy is restored after a server sync.

So basically with sync remote deletions on, Move to trash and Delete practically do the same thing, except Delete trashes local copy immediately.

Why is this so complicated and confusing? No normal user is going to properly keep track of this kind of behavior, not to mention Delete doesn't actually do what most people think it does (permanent removal).

The behavior really should just be:
1. **Move to Trash:** Immediately move local copy to Android trash and server copy to Immich trash. A restore on the Android device will immediately move local copy out of Android trash and move server copy back to timeline. A restore that's triggered on a different device or via the web will cause the Android device to restore the local copy the next time it does a server sync.
2. **Delete:** Immediate deletion from both local and server after a confirmation dialog shown to the user. No restore possible. This is no joke guys. If you don't want users getting confused and accidentally getting their nudes and other private photos leaked, this is how you make it clear to the user. Any other behavior is misleading. **Delete means delete!**

**Comment by hadshaw21** [comment]
iOS user here, and I think for now, a fix that could greatly improve quality of life without entirely changing the way that the delete functions work is to simply change the order of the buttons when selecting multiple photos.

Currently when selecting one/multiple photos, the first button that shows up is "Move to Trash", to get to "Delete" and "Delete from Device" you have to scroll all the way to the right, which is better than nothing but if you're not paying attention and are just trying to delete some photos, you'll forget to do, and just hit "Move to Trash" instead as its right there, which ends up leaving the photos on your device, which will eventually re-upload them.

As someone who comes from Google Photos, I'm already used to the secondary iOS delete pop-up, but this only happens when you scroll over and press "Delete". Maybe I am an outlier, but I feel it would be way more useful to have the **Combined** "Delete" button as the primary one shown. Definitely still keep the "Move to Trash" and "Delete from Device" as options, but I feel they are much less useful and could be moved to the end instead, but that way we are not removing or changing and functionality, simply changing the order for ease of use.

What shows up when you first select item/s:
<img width="490" height="345" alt="Image" src="https://github.com/user-attachments/assets/1b33702e-0d65-452f-8f3e-0e7fb34d8c6f" />

What shows when you scroll all the way to the right of the toolbar:
<img width="490" height="349" alt="Image (1)" src="https://github.com/user-attachments/assets/f05fefb9-64fa-4a9a-b95e-96bcb8c0e168" />

**Reply by hadshaw21** [reply]
Also just wondering if rather than making a separate feature request for iOS, could we just update the title of this one to reflect all Mobile apps?

**Reply by hadshaw21** [reply]
This is how Google Photos does it for reference:

What first shows up:
<img width="800" height="226" alt="image" src="https://github.com/user-attachments/assets/ea7ccfa7-4956-42d2-9877-6a5455645721" />

When you scroll all the way to the right:
<img width="1290" height="323" alt="image" src="https://github.com/user-attachments/assets/687e20ba-f963-41c6-96ad-d1a35d83f247" />

**Comment by FraYoshi** [comment]
I was constantly finding the deleted photos again reupped, and the photo deleted on Immich were still on the phone, that was super confusing, and I thought Immich was just broken. Apparently, it was supposed to do that...
Interesting how we need to scroll all the way to the bottom, but not totally to the bottom, for each and every photo we wish to delete....
When I delete from the Immich app, I mean to obliterate the photo from existence, not have it deleted half-way 😅 I would gladly switch the "Delete" and the "Move to Trash", as I think that when someone deletes it from the gallery, they are actually deleting the picture, not trying to save space.... on the server but not on the phone, and ONLY TEMPORARILY until 30 days pass, and it's upped again 😅 which does not even make sense, as it the trash is still taking up space, so it's kinda unuseful 😅😅

**Comment by hadshaw21** [comment]
Am I tripping or has this been pushed in latest iOS app version (3.2)? I can’t find it anywhere in patch notes for last few versions and this feat is still open, curious, but appreciated.

**Comment by TitaniumCoder477** [comment]
I took a bunch of photos at my nephew's birthday party yesterday. When taking pictures of 5 year old, you have to take a lot to get one good one! So this morning I selected the ones I didn't want to keep but had to select them twice and delete them twice, once using the "Delete from device" button and the other using the "Trash" button.

This is a task I perform very often because usually when I delete something from my device, I don't want it on Immich either (and vise versus).

It would be really, really helpful to have a single button to handle both actions so that my finger effort (especially in multi selecting lots of unwanted pictures) is not duplicated. **It appears this button is available to some people already, but it does not appear available to me on the Android app.**

(Or even better, the solution that another user in this convo suggested: https://github.com/immich-app/immich/discussions/14441#discussioncomment-15280102)

If you develop this button and feel the need to gate it with a warning (since it could result in the total loss of one or more assets), please make that warning something I can disable in settings. I would personally turn that off because I rarely need to confirm even an action such as this that I have intentionally invoked.
